src/openzaak/components/zaken/brondatum.py

```diff
@@ -36,25 +37,32 @@ def calculate(self) -> Union[None, date]:
         if isinstance(archiefactietermijn, str):
             archiefactietermijn = parse_relativedelta(archiefactietermijn)
 
-        brondatum_archiefprocedure = resultaattype.brondatum_archiefprocedure
-        afleidingswijze = brondatum_archiefprocedure["afleidingswijze"]
-        datum_kenmerk = brondatum_archiefprocedure["datumkenmerk"]
-        objecttype = brondatum_archiefprocedure["objecttype"]
-        procestermijn = brondatum_archiefprocedure["procestermijn"]
-        # if loose-fk-field - convert to relative-delta
-        if isinstance(procestermijn, str):
-            procestermijn = parse_relativedelta(procestermijn)
-
-        # FIXME: nasty side effect
-        orig_value = self.zaak.einddatum
-        self.zaak.einddatum = self.datum_status_gezet.date()
-        brondatum = get_brondatum(
-            self.zaak, afleidingswijze, datum_kenmerk, objecttype, procestermijn
-        )
-        self.zaak.einddatum = orig_value
+        # if zaak.startdatum_bewaartermijn is filled -> use as brondatum
+        if self.zaak.startdatum_bewaartermijn:
+            brondatum = self.zaak.startdatum_bewaartermijn
+
+        else:
+            # calculate brondatum using resultaattype.brondatum_archiefprocedure
+            brondatum_archiefprocedure = resultaattype.brondatum_archiefprocedure
+            afleidingswijze = brondatum_archiefprocedure["afleidingswijze"]
+            datum_kenmerk = brondatum_archiefprocedure["datumkenmerk"]
+            objecttype = brondatum_archiefprocedure["objecttype"]
+            procestermijn = brondatum_archiefprocedure["procestermijn"]
+            # if loose-fk-field - convert to relative-delta
+            if isinstance(procestermijn, str):
+                procestermijn = parse_relativedelta(procestermijn)
+
+            # FIXME: nasty side effect
+            orig_value = self.zaak.einddatum
+            self.zaak.einddatum = self.datum_status_gezet.date()
+            brondatum = get_brondatum(
+                self.zaak, afleidingswijze, datum_kenmerk, objecttype, procestermijn
+            )
+            self.zaak.einddatum = orig_value
```

Comment:
I haven't looked into this, but would it take a lot of effort to refactor this to remove this zaak.einddatum mutation?